src/ray/object_manager/object_manager.cc

```diff
@@ -409,6 +410,59 @@ void ObjectManager::PushLocalObject(const ObjectID &object_id, const NodeID &nod
                      std::move(release_chunk_callback));
 }
 
+void ObjectManager::PushSpilledObject(const ObjectID &object_id, const NodeID &node_id,
+                                      const std::string &spilled_url) {
+  // SpilledObject::CreateSpilledObject does synchronous IO; schedule it off
+  // main thread.
```

Comment:
RAY_CHECK(RayConfig::instance().is_external_storage_type_fs())

Reply:
added the check [here ](https://github.com/ray-project/ray/pull/16364/files#diff-1b792600903619ac4375f3eaf4cfe68f30cb8df813128d2a6fbee968a656554bR341)as it seems more appropriate

Reply:
We could still RAY_CHECK here to catch any logic errors.